src/resolvers/intermine/bio-entity.ts

```diff
@@ -0,0 +1,32 @@
+import { DataSources, IntermineAPI } from '../../data-sources/index.js';
+import { KeyOfType } from '../../utils/index.js';
+import { SubfieldResolverMap } from '../resolver.js';
+import { annotatableFactory } from './annotatable.js';
+
+
+export const bioEntityFactory =
+(sourceName: KeyOfType<DataSources, IntermineAPI>): SubfieldResolverMap => ({
+    ...annotatableFactory(sourceName),
+    organism: async (bioEntity, _, { dataSources }) => {
+        return dataSources[sourceName].getOrganism(bioEntity.organismTaxonId)
+            // @ts-ignore: implicit type any error
+            .then(({data: results}) => results);
+    },
+    strain: async (bioEntity, _, { dataSources }) => {
+        return dataSources[sourceName].getStrain(bioEntity.strainIdentifier)
+            // @ts-ignore: implicit type any error
+            .then(({data: results}) => results);
+    },
+    locations: async (bioEntity, { page, pageSize }, { dataSources }) => {
+        const args = {sequenceFeature: bioEntity, page, pageSize};
```

Comment:
This bugs me having sequenceFeature: bioEntity inside the BioEntity resolver. Is this correct?

Reply:
@sammyjava I agree. It's because the `getLocations` method (src/data-sources/intermine/api/get-locations.ts) expects the `sequenceFeature` named parameter. It was already coded that way so I didn't change it. Since locations are a bio entity property and all sequence features are bio entities I think it makes sense to change the named parameter.